code/go/0chain.net/smartcontract/dbs/event/challenge.go

```diff
@@ -28,6 +28,19 @@ type Challenge struct {
 	ExpiredN       int              `json:"expired_n" gorm:"-"`
 }
 
+func (edb *EventDb) CleanDataFromChallengesAndRewardProviders() error {
+	if err := edb.Store.Get().Delete(&Challenge{}).Error; err != nil {
+		return err
+	}
+	return edb.Store.Get().Delete(&RewardProvider{}).Error
```

Comment:
if you want to delete all the data from the table use TRUNCATE it's much faster, since it doesn't use redo logs